rust-src/wallet_library/src/statement.rs

```diff
@@ -0,0 +1,274 @@
+use concordium_base::{
+    curve_arithmetic::Curve,
+    id::{id_proof_types::AtomicStatement, types::*},
+    web3id::{CredentialStatement, Request},
+};
+
+const IDENTITY_RANGE_TAGS: &[AttributeTag] = &[AttributeTag(3), AttributeTag(9), AttributeTag(10)]; // dob, idDocIssuedAt, idDocExpiresAt
+const IDENTITY_SET_TAGS: &[AttributeTag] = &[
+    AttributeTag(4),
+    AttributeTag(5),
+    AttributeTag(6),
+    AttributeTag(8),
+]; // Country of residence, Nationality, IdDocType, IdDocIssuer
+
+fn is_iso8601(date: &str) -> bool {
+    println!("{}", date);
+    chrono::NaiveDate::parse_from_str(date, "%Y%m%d").is_ok()
+}
+
+fn is_iso3166_alpha_2(code: &str) -> bool { rust_iso3166::from_alpha2(code).is_some() }
+
+fn is_iso3166_2(code: &str) -> bool { rust_iso3166::iso3166_2::from_code(code).is_some() }
+
+pub trait WalletCheck {
+    fn satisfies_wallet_restrictions(&self) -> bool;
+}
```

Comment:
I would have expected a trait more in line with
```rust
pub trait AcceptableRequest {
    type Error;
    fn acceptable_request(&self, wallet: &WalletConfig) -> Result<(), Error>;
}
```

and you'd have an enum WalletConfig.

The `Error` would be an enum as well that could be used to display something meaningful to either the user or dapp.

I'm imagining the scenario where a dapp asks for a web3id proof of the mobile wallet. What will the experience be?

Reply:
Yeah, I've have changed it to use Result, and added a `WalletConfig` that allows the user to specify tags that are allowed for range and set, and to provide a function for "custom" attribute verification.